articles/unofficial-faq.html

```diff
@@ -526,23 +526,23 @@ <h3>
 
 <a class="anchor" id="function-inline"></a>
 <h3>
-	Does the standard Go compiler support function inline?
+	Does the standard Go compiler support inline functions?
 </h3>
 
 <p>
-Yes, the standard Go compiler supports function inline.
-The compiler will automatically inline some very short functions which satisfying some requirements.
+Yes, the standard Go compiler supports inline functions.
+The compiler will automatically inline some very short functions which satisfy certain requirements.
 The specific inline requirements may change from version to version.
 </p>
 
 <div>
-Currently (Go Toolchain 1.15), for the standard Go compiler,
+Currently (Go Toolchain 1.15), for the standard Go compiler:
 <ul>
-<li>there is no explicit ways to specify which functions should be inlined in user programs.</li>
-<li>although <code>-gcflags "-l"</code> build option can prevent any functions being inline,
-	there are not formal ways to avoid specified functions being inline in user programs.
+<li>There are no explicit ways to specify which functions should be inlined in user programs.</li>
+<li>The <code>-gcflags "-l"</code> build option disables inlining globally, which will prevent all functions from being inline expanded.
+<li>In user programs, there are not formal ways to prevent inlining of particular functions.
 	You can add a line <code>//go:noinline</code> directive before a function declaration to
-	avoid the function being inline now, but this way is not guaranteed to work in the future.
+	avoid the function being inlined now, but this way is not guaranteed to work in the future.

```

Comment:
You often say things "are not guaranteed to exist in future Go versions." Could you explain more of what you mean here, or  maybe in a meta-FAQ?

Reply:
That makes sense. So presumably, the removal of that feature is under discussion somewhere in the official go project?

Reply:
No. It is just that //go:xxx directive is just a design for the standard Go compiler. It is not a formal Go feature. The standard Go compiler might change the design later.